### ISSUE TITLE
Return type of unsubclassed `Enum.__new__` and friends is unknown

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#4010]
**Describe the bug**
When attempting to instantiate an `Enum` from the stdlib (be it `Enum` itself or `StrEnum`, etc.) that has not been subclassed, the return type is unknown.

(Incidentally, Pylance reports that the type of `enum_member` is `Any`, but that might be because it bundles an older version of Pyright.  [Is there a way to get Pylance to use the Pyright extension instead?])

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That the type of `enum_member` would be `Enum`.

**Screenshots or Code**

```python
# pyright: strict

from enum import Enum
from typing import TypeVar

TEnum = TypeVar("TEnum", bound=Enum)


def foo(enum_cls: type[TEnum], enum_value: object) -> TEnum:
    enum_member = enum_cls(enum_value)
    reveal_type(enum_member)  # Unknown
    return enum_member  # Error: Return type is unknown
```

**VS Code extension or command-line**
Either, 1.1.274

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.275, which I just published. It will also be included in a future release of pylance.
